### PR TITLE
Implement `only` for `TracedRArray`

### PR DESCRIPTION
**Pull request by mofeing**
@wsmoses In Julia, a 0-dim array is not (programmatically) equal to a scalar, but we're treating as such in Reactant. We should be more careful about this, but meanwhile, I'm using this method implementation to keep compatibility between Reactant and non-Reactant code.

### CODE REVIEW

**Comment by wsmoses** [comment]
Yeah the issue is overloading or an acutual AbstractFloat requires a new type in addition to the RArray types, so we've been using the 0-dim for this for ease. We definitely should figure out a good general way to handle

File: src/TracedRArray.jl [hunk 1]
```diff
@@ -29,6 +29,8 @@ function Base.show(io::IOty, X::TracedRArray{T,N}) where {T,N,IOty<:Union{IO,IOC
     return print(io, X.mlir_data, ")")
 end
 
+Base.only(A::TracedRArray) = A
```
Comment (wsmoses) [inline src/TracedRArray.jl:32]: Can we have this only apply for 0-dim arrays?

**Comment by mofeing** [comment]
An alternative could be to implement something like `TrackedReal` in ReverseDiff.jl, but in the MLIR-side is still a 0-dim tensor.

```julia
struct TracedNumber{T} <: Number
    mlir_data
end
```

File: src/TracedRArray.jl (hunk 1 above)
Comment (mofeing) [reply src/TracedRArray.jl]: done!

**Review by wsmoses — APPROVED** [review]

**Pull request merged**